### PR TITLE
[Customer Portal][FE][Web] Refactor Change Request State Filtering to Use Metadata-Driven Resolution

### PR DESCRIPTION
**Pull request by dileepapeiris**
### Description

This pull request refactors how change request (CR) state filtering is handled throughout the customer portal’s dashboard and operations features. The main improvement is replacing static, hardcoded CR state ID lists with new utility functions that dynamically resolve the correct state IDs from project metadata. This makes the codebase more robust to backend changes and improves maintainability. Additionally, related tests and UI components have been updated to use these new utilities.

**Change Request State Filtering Refactor:**

* Replaced all hardcoded CR state ID arrays (e.g., `CR_ACTION_REQUIRED_STATE_KEYS`, `OUTSTANDING_CHANGE_REQUEST_STATE_IDS`, `ALLOWED_CHANGE_REQUEST_STATE_IDS`) with utility functions (`resolveActionRequiredCrStateIds`, `resolveOutstandingCrStateIds`, `resolveAllowedCrStateIds`, etc.) that compute state IDs based on current project metadata. This ensures the frontend always uses the correct states as defined by the backend. [[1]](diffhunk://#diff-cda7965095932ea3788027824b371a97c999333cb8b66e0963e987de7d2598c8L52-L55) [[2]](diffhunk://#diff-94f6a8a1ed1fdbada17db5f363f063a846fdd08bd092dc3a3970733c4a474d5dL36-R37) [[3]](diffhunk://#diff-58687b500ef8553bdd96f52b4620b6c38d7fbb91bba51451fc3a399bfc5cdbf6L55-R59)

* Updated all usages of CR state ID lists in data fetching hooks and filters (e.g., in `DashboardItemsPage.tsx`, `OperationsPage.tsx`, `ChangeRequestsPage.tsx`) to use the new resolver functions, and to properly handle cases where metadata is not yet loaded. [[1]](diffhunk://#diff-cda7965095932ea3788027824b371a97c999333cb8b66e0963e987de7d2598c8L120-R125) [[2]](diffhunk://#diff-cda7965095932ea3788027824b371a97c999333cb8b66e0963e987de7d2598c8L151-R155) [[3]](diffhunk://#diff-cda7965095932ea3788027824b371a97c999333cb8b66e0963e987de7d2598c8L239-R243) [[4]](diffhunk://#diff-58687b500ef8553bdd96f52b4620b6c38d7fbb91bba51451fc3a399bfc5cdbf6R109-R113) [[5]](diffhunk://#diff-58687b500ef8553bdd96f52b4620b6c38d7fbb91bba51451fc3a399bfc5cdbf6L144-R165) [[6]](diffhunk://#diff-58687b500ef8553bdd96f52b4620b6c38d7fbb91bba51451fc3a399bfc5cdbf6L175-R185) [[7]](diffhunk://#diff-ddbfee321640578ea9b461da5bfcef82d7bdbdf644ee5bfe588b1644b6aafd5fL118-R119) [[8]](diffhunk://#diff-ddbfee321640578ea9b461da5bfcef82d7bdbdf644ee5bfe588b1644b6aafd5fL134-R137) [[9]](diffhunk://#diff-ddbfee321640578ea9b461da5bfcef82d7bdbdf644ee5bfe588b1644b6aafd5fR154)

**Testing and Utility Improvements:**

* Added comprehensive unit tests for the new resolver functions, ensuring they handle all relevant edge cases and state label exclusions. Updated tests for `buildChangeRequestSearchRequest` to verify integration with the new state ID resolution logic. [[1]](diffhunk://#diff-2ac95970e81f64fec9d4764542bf7ae312356fbf49d352b1bc684e20a467bd0cR26-R30) [[2]](diffhunk://#diff-2ac95970e81f64fec9d4764542bf7ae312356fbf49d352b1bc684e20a467bd0cR52-R180)

**UI and Component Updates:**

* Updated the CR calendar legend to accept a dynamic list of states, allowing it to reflect the actual states available in the project metadata. [[1]](diffhunk://#diff-a160affd3ad8642442166f018c9283a88b288e9ad82652b9c8fb8eae7e71f197R54-R57) [[2]](diffhunk://#diff-a160affd3ad8642442166f018c9283a88b288e9ad82652b9c8fb8eae7e71f197L396-R403) [[3]](diffhunk://#diff-a160affd3ad8642442166f018c9283a88b288e9ad82652b9c8fb8eae7e71f197L414-R417) [[4]](diffhunk://#diff-bc7e5e2dfd0731d1e8efcc62067206b44638f6703e561d7808f0730c5ce46802R172) [[5]](diffhunk://#diff-ddbfee321640578ea9b461da5bfcef82d7bdbdf644ee5bfe588b1644b6aafd5fR382)

**Bug Fixes and Minor Adjustments:**

* Ensured CR-related UI elements and queries are only enabled when relevant metadata is loaded, preventing errors due to missing state information. [[1]](diffhunk://#diff-cda7965095932ea3788027824b371a97c999333cb8b66e0963e987de7d2598c8L151-R155) [[2]](diffhunk://#diff-ddbfee321640578ea9b461da5bfcef82d7bdbdf644ee5bfe588b1644b6aafd5fL134-R137) [[3]](diffhunk://#diff-ddbfee321640578ea9b461da5bfcef82d7bdbdf644ee5bfe588b1644b6aafd5fR154) [[4]](diffhunk://#diff-58687b500ef8553bdd96f52b4620b6c38d7fbb91bba51451fc3a399bfc5cdbf6L144-R165)

* Simplified CR and service request counts in the dashboard to use the new outstanding count directly, improving accuracy and readability.

Overall, this PR makes CR state handling more reliable and adaptable to backend changes, while improving test coverage and user interface flexibility.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed Change Request query execution to prevent undefined state values from being passed
  * Improved state filtering to depend on available filter metadata

* **Refactor**
  * Replaced hardcoded state ID arrays with dynamic resolution from filter metadata
  * Updated Change Request query construction to use metadata-driven state selection
  * Calendar view now reflects available change request states from metadata

### CODE REVIEW

File: apps/customer-portal/webapp/src/features/operations/utils/operationsPages.ts [hunk 1]
```diff
@@ -75,20 +72,124 @@ export function formatOperationsOverviewChangeRequestsSubtitle(
...
+ * @returns Array of numeric state IDs, or `undefined` if metadata is not yet loaded.
+ */
+export function resolveClosedCrStateIds(
+  changeRequestStates: MetadataItem[] | undefined,
+): number[] | undefined {
+  if (!changeRequestStates) return undefined;
+  return changeRequestStates
+    .filter((s) => CLOSED_CR_STATE_LABELS.has(s.label))
+    .map((s) => Number(s.id));
+}
```
[the remark this answers is not in the document]
Comment (dileepapeiris) [reply apps/customer-portal/webapp/src/features/operations/utils/operationsPages.ts:180]: We used enum here

File: apps/customer-portal/webapp/src/features/operations/pages/ChangeRequestsPage.tsx [hunk 2]
```diff
@@ -115,8 +115,8 @@ export default function ChangeRequestsPage(): JSX.Element {
   });
 
   const changeRequestSearchRequest = useMemo(
-    () => buildChangeRequestSearchRequest(filters, searchTerm, outstandingOnly, actionRequired, scheduledOnly),
-    [searchTerm, filters, outstandingOnly, actionRequired, scheduledOnly],
+    () => buildChangeRequestSearchRequest(filters, searchTerm, outstandingOnly, actionRequired, scheduledOnly, filterMetadata?.changeRequestStates),
+    [searchTerm, filters, outstandingOnly, actionRequired, scheduledOnly, filterMetadata?.changeRequestStates],
   );
```
[the remark this answers is not in the document]
Comment (dileepapeiris) [reply apps/customer-portal/webapp/src/features/operations/pages/ChangeRequestsPage.tsx]: Updated

**Review by Rashmika998 — APPROVED** [review]
